### ISSUE TITLE
Error handling when xdg-user-dir is not found

### ISSUE THREAD

**Issue by gjj** [juliangruber/downloads-folder#10]
I was using this package as part of an integration test in a containerized environment, and I found out that my container does not have `xdg-user-dir`. Because of that, the following line causes my pipeline to exit prematurely:

```
dir = execSync('xdg-user-dir DOWNLOAD', { stdio: [0, 3, 3] });
```

Just wanted to check what does the `[0, 3, 3]` mean? Because I can't find it in the Node.js docs: https://nodejs.org/api/child_process.html#child_process_options_stdio

**Comment by juliangruber** [comment]
The file descriptors `0` and `3` stand for `stdin` and `ignore`. This is listed on the docs page you linked, but admittedly not very prominent. We should make that more explicit.

In the current logic this means that `xdg-user-dir` will inherit `stdin` but ignore `stdout` and `stderr`. If I understand this correctly, now I wonder whether this code ever worked. For me `execSync` always throws if `stdout` is set to `ignore`, so this has to be fixed.

I do wonder however why this makes your pipeline exit, because the code is wrapped in a `try/catch` and shouldn't ever throw.

**Comment by juliangruber** [comment]
I published `2.0.0` with a fix, can you try this please?

**Comment by gjj** [comment]
@juliangruber thanks for the clarifications, I missed out the part where it says `fd` 3 and up means `ignore`.

And thanks again for the fix - it worked! Cheers! :smile:
